src/Paprika.Benchmarks/SlottedArrayBenchmarks.cs

```diff
@@ -191,6 +227,43 @@ public int EnumerateNibble(byte nibble)
         return length;
     }
 
+    [Benchmark]
+    public bool Defragmentation()
+    {
+        var map = new SlottedArray(new Span<byte>(_defragmentMap, Page.PageSize));
+        Span<byte> value = stackalloc byte[1];
+
+        // Insert additional key to trigger defragmentation
+        var additionalKey = Keccak.Zero;
+        value[0] = KeyCount;
+
+        return map.TrySet(NibblePath.FromKey(additionalKey), value);
+    }
+    
+    [Benchmark]
+    [Arguments(1)]
+    [Arguments(25)]
+    [Arguments(50)]
+    [Arguments(96)]
+    public bool Defragmentation_Large(int count)
+    {
+        var map = new SlottedArray(new Span<byte>(_defragmentMap, Page.PageSize));
```

Comment:
This is wrong I believe. You should always copy the payload first, otherwise you'll be overwriting the same memory. This is what I meant when discussing the `OperationsPerInvoke` so that you amortize the sets. The first step is to copy it to a working memory (a separate aligned page), then perform a few deletes & sets. My take would be to set at least 2 times or more so that the initial copy is not the major hit.